filestore/filepath.go

```diff
@@ -0,0 +1,520 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package filestore
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"strings"
+)
+
+type FileType string
+
+type FileStoreType string
+
+const (
+	Memory     FileStoreType = "MEMORY"
+	FileSystem FileStoreType = "LOCAL_FILESYSTEM"
+	Azure      FileStoreType = "AZURE"
+	S3         FileStoreType = "S3"
+	GCS        FileStoreType = "GCS"
+	HDFS       FileStoreType = "HDFS"
+)
+
+const (
+	Parquet FileType = "parquet"
+	CSV     FileType = "csv"
+	DB      FileType = "db"
+)
+
+const (
+	GSPrefix         = "gs://"
+	S3Prefix         = "s3://"
+	S3APrefix        = "s3a://"
+	AzureBlobPrefix  = "abfss://"
+	HDFSPrefix       = "hdfs://"
+	FileSystemPrefix = "file://"
+)
+
+var ValidSchemes = []string{
+	GSPrefix, S3Prefix, S3APrefix, AzureBlobPrefix, HDFSPrefix, FileSystemPrefix,
+}
+
+func (ft FileType) Matches(file string) bool {
+	ext := GetFileExtension(file)
+	return FileType(ext) == ft
+}
+
+func IsValidFileType(file string) bool {
+	for _, fileType := range []FileType{Parquet, CSV, DB} {
```

Comment:
Use AllFileTypes from comment above, these sort of things tend to get out of sync